fix(sidebar): wrap long menu item label in sidebar

ciiay opened the pull request:
## Description

- Modified Backstage SidebarItem styles to allow long menu item labels to wrap instead of being cut off.
- Refactored code to simplify the codebase and improve sidebar menu display, especially on smaller screens.

[Figma reference](https://www.figma.com/design/Gp6QQ0wtRO3RSCphRZIdjQ/RHDH-Theme?node-id=908-9509&p=f&t=8n10OTFioWinXaMl-0)

## Which issue(s) does this PR fix

- Fixes [RHIDP-6010](https://issues.redhat.com/browse/RHIDP-6010)

## PR acceptance criteria

Please make sure that the following steps are complete:

- [ ] GitHub Actions are completed and successful
- [ ] Unit Tests are updated and passing
- [ ] E2E Tests are updated and passing
- [ ] Documentation is updated if necessary (requirement for new features)
- [x] Add a screenshot if the change is UX/UI related

## Screenshot
![image](https://github.com/user-attachments/assets/4ab81d57-da95-4b1a-af4f-b0fcef7e662b)

ciiay commented:
/test e2e-tests

ciiay commented:
/test e2e-tests

ciiay commented:
/test e2e-tests

ciiay commented:
/test e2e-tests

ShiranHi commented:
Thank you, @ciiay, for this change! I have two questions:  

1. Can we ensure the text at the second and third levels is aligned?  
<img width="235" alt="image01" src="https://github.com/user-attachments/assets/23c3c725-c489-45f7-882e-8b0d4b632722" />

2. Is there a reason for the large padding below the second level?
<img width="256" alt="image02" src="https://github.com/user-attachments/assets/70bb25e5-7022-406b-bcea-ffd8859782d7" />

ciiay commented:
> 1. Can we ensure the text at the second and third levels is aligned?

Fixed it, thanks.
![image](https://github.com/user-attachments/assets/3895aa31-afaa-4bea-95d3-03541577b531)

> 2. Is there a reason for the large padding below the second level?

It is not a gap, just end of the menu list. If I collapse the Administration dropdown it'll be empty space only. I moved "Docs" out of its parent menu item so you can see clearer how it is.
![image](https://github.com/user-attachments/assets/a23951ff-9053-4416-b7f0-04af2a3a15e4)

ciiay commented:
/test e2e-tests

ciiay commented:
/test e2e-tests

its-mitesh-kumar commented:
Tested the PR on cluster . 
With below config 
```
dynamicPlugins:
      frontend:
        default.main-menu-items:
          menuItems:
            default.home:
              title: Home
              icon: home
            default.list:
              title: References
              icon: bookmarks
            default.my-group:
              parent: default.list
            default.learning-path:
              title: asdfghjgfdsaDZFGHJBJHGZESRDTFYGUKHJDSADFZXGCHVNDHBGSZ≈CVBNMDFGHJKLHFGHJKL;KJHFGCUYYLIUYTKTDTGIKJH,GFDJFGDCGFILUYUYTRDXC,JHGBFDCJKHGCFGCJKKH
              parent: default.list
            default.homepage:
              title: HomePage 123
              icon: home
              parent: default.home
            default.create:
              title: Long Createing Title abcdefghjkllooiuysdfgvndfbhvjsbfs fcbdhjgvbhjkdsbchjdsbvjhdsb
              icon: add
              parent: default.home


```
![Screenshot 2025-03-21 at 5 12 56 PM (2)](https://github.com/user-attachments/assets/a0ba5218-487a-41ee-8e7a-5f85ff636efb)

its-mitesh-kumar reviewed (commented):
See the comment https://github.com/redhat-developer/rhdh/pull/2566#issuecomment-2743145272

ciiay commented:
Hi @its-mitesh-kumar , thanks for the review. 
>A. What I observed Littlelonglearningpathweneedtomange is getting trimmed followed by dots but same words with space ie. Little long learning path we need to mange  getting wrapped to multiple lines .

Fixed.

>B. Do we support grand child in menu items , everything seems fine , only icon is not visible if we have some grand child of menu item .

This is designed like this, the third level menu items don't have an icon. 

Hi @ShiranHi , I noticed that the previous Figma link you shared now always redirects to the cover link. I saw you update it in Bug ticket 188, and I attached that link to the PR description. It used to work before but now whatever I do it only goes to the cover page. Can you share the design page of sidebar menu here? And I wonder if there's a better way to navigate from the cover page to a specific page. I tried to figure it out from this [RHDH core](https://www.figma.com/files/1210702722894069364/project/109099485?fuid=1099360402350253917) page, but I couldn't find it anywhere 😭

ShiranHi commented:
> Hi @ShiranHi , I noticed that the previous Figma link you shared now always redirects to the cover link. I saw you update it in Bug ticket 188, and I attached that link to the PR description. It used to work before but now whatever I do it only goes to the cover page. Can you share the design page of sidebar menu here? And I wonder if there's a better way to navigate from the cover page to a specific page. I tried to figure it out from this [RHDH core](https://www.figma.com/files/1210702722894069364/project/109099485?fuid=1099360402350253917) page, but I couldn't find it anywhere 😭

Oh, sorry for that! Here’s the [Figma link](https://www.figma.com/design/Gp6QQ0wtRO3RSCphRZIdjQ/RHDH-Theme?node-id=1299-5160&t=WQlxmbq13TFk1Ap1-1) to the menu page. 

You can try scrolling through the page list on the left in Figma to find all the components:

https://github.com/user-attachments/assets/28d881e9-6f9d-452e-b893-3cc1e74a98d8

ciiay commented:
Thank you @ShiranHi for the recording 💯  I must have over looked "Main navigation" from the Pages list. I was a bit confused by the wording "Pages" and thought sidebar didn't below to any pages. And tried searching with "Sidebar" but got no luck 😅

its-mitesh-kumar reviewed (commented):
I verified on cluster up-to 3 words ,it's working as per expected .

its-mitesh-kumar commented:
Verified on cluster working as per expected for 3-4 words long menu item , screenshot with config for reference 
```
dynamicPlugins:
      frontend:
        default.main-menu-items:
          menuItems:
            default.home:
              title: Home
              icon: home
            default.list:
              title: References
              icon: bookmarks
            default.my-group:
              title: Component, System, Group, Resource
              parent: default.list
            default.learning-path:
              title: Little long learning path we need to mange
              parent: default.list
            default.create:
              title: Long Creating
              icon: add
              parent: default.home
            default.create1:
              title: Long Creating Title
              icon: add
              parent: default.home
            default.create2:
              title: Long Creating Title Lets
              icon: add
              parent: default.home
            default.create3:
              title: Long Creating  Title Lets see 
              icon: add
              parent: default.home
            default.homepage:
              title: Littlelonglearningpathweneedtomange
              icon: home
              parent: default.home
            default.learning-path2:
              title: Little long 
              parent: default.homepage
            default.learning-path3:
              title: Little long learning 
              parent: default.homepage
            default.learning-path4:
              title: Little long learning path 
              parent: default.homepage
            default.learning-path5:
              title: Little long learning path we 
              parent: default.homepage
            default.learning-path6:
              title: Little long learning path we need 
              parent: default.homepage
            default.learning-path7:
              title: Little long learning path we need to mange
              parent: default.homepage
```
![Screenshot 2025-03-24 at 6 23 21 PM (2)](https://github.com/user-attachments/assets/d0b22973-c081-4994-9b4f-3b5b7d250dfc)

its-mitesh-kumar commented:
/lgtm
/approve
